python/ray/serve/_private/replica.py

```diff
@@ -484,26 +479,36 @@ async def invoke_single(self, request_item: Query) -> Tuple[Any, bool]:
 
         return result, success
 
-    async def reconfigure(self, user_config: Any):
+    async def reconfigure(self, deployment_config: DeploymentConfig):
         async with self.rwlock.writer_lock:
-            self.user_config = user_config
-            self.version = DeploymentVersion(
-                self.version.code_version, user_config=user_config
+            user_config_changed = False
+            if (
```

Comment:
nit: should we directly `if self.deployment_config.user_config != deployment_config.user_config`

Reply:
When the replica is first initialized, deployment_config is None.